src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,115 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector((state) => state.todos);
+  const filter = useAppSelector((state) => state.filter);
+
+  const currentTodo = useAppSelector((state) => state.currentTodo);
+
+  const filteredTodos: Todo[] | undefined = useMemo(() => {
+    const newArray = todos?.filter((todo) => {
+      const lowerCase = todo.title.toLowerCase();
+      const filterLowerCase = filter.query?.toLowerCase() || '';
+
+      switch (filter.status) {
+        case 'active':
+          return !todo.completed && lowerCase.includes(filterLowerCase);
+        case 'completed':
+          return todo.completed && lowerCase.includes(filterLowerCase);
+        default:
+          return lowerCase.includes(filterLowerCase);
+      }
+    });
+
+    return newArray;
+  }, [filter]);
```

Comment:
No need to filter in case status is all and query is empty! It my cause performence issue
Also to avoid using optional chaining in code below you can use such approach - and it alway will be an array
```
(todos || []).filter
```